test/conformance/helpers/broker_tracing_test_helper.go

```diff
@@ -0,0 +1,489 @@
+	ingressHost := fmt.Sprintf("%s-broker.%s.svc.%s", brokerName, client.Namespace, domain)
+	ingressChanHost := fmt.Sprintf("%s-kne-ingress-kn-channel.%s.svc.%s", brokerName, client.Namespace, domain)
+	triggerChanHost := fmt.Sprintf("%s-kne-trigger-kn-channel.%s.svc.%s", brokerName, client.Namespace, domain)
+	filterHost := fmt.Sprintf("%s-broker-filter.%s.svc.%s", brokerName, client.Namespace, domain)
+	loggerTriggerPath := fmt.Sprintf("/triggers/%s/%s/%s", client.Namespace, loggerTrigger.Name, loggerTrigger.UID)
+	transformerTriggerPath := fmt.Sprintf("/triggers/%s/%s/%s", client.Namespace, transformerTrigger.Name, transformerTrigger.UID)
+	loggerSVCHost := fmt.Sprintf("%s.%s.svc.%s", loggerPodName, client.Namespace, domain)
+	transformerSVCHost := fmt.Sprintf("%s.%s.svc.%s", eventTransformerPod.Name, client.Namespace, domain)
+
+	// This is very hard to read when written directly, so we will build piece by piece.
+
+	// Steps 17-18: 'logger' event being sent to the 'transformer' Trigger.
+	loggerEventSentFromTrChannelToTransformer := tracinghelper.TestSpanTree{
+		Note: "17. Broker TrChannel sends the event to the Broker Filter for the 'transformer' trigger.",
+		Kind: model.Client,
+		Tags: map[string]string{
+			"http.method":      "POST",
+			"http.status_code": "202",
+			"http.url":         fmt.Sprintf("http://%s%s", filterHost, transformerTriggerPath),
+		},
+		Children: []tracinghelper.TestSpanTree{
+			{
+				Note: "18. Broker Filter for the 'transformer' trigger receives the event from the Broker TrChannel. This does not pass the filter, so this 'branch' ends here.",
+				Kind: model.Server,
+				Tags: map[string]string{
+					"http.method":      "POST",
+					"http.status_code": "202",
+					"http.host":        filterHost,
+					"http.path":        transformerTriggerPath,
+				},
+			},
+		},
+	}
+
+	// Steps 19-22: 'logger' event being sent to the 'logger' Trigger.
+	loggerEventSentFromTrChannelToLogger := tracinghelper.TestSpanTree{
+		Note: "19. Broker TrChannel sends the event to the Broker Filter for the 'logger' trigger.",
+		Kind: model.Client,
+		Tags: map[string]string{
+			"http.method":      "POST",
+			"http.status_code": "202",
+			"http.url":         fmt.Sprintf("http://%s%s", filterHost, loggerTriggerPath),
+		},
+		Children: []tracinghelper.TestSpanTree{
+			{
+				Note: "20. Broker Filter for the 'logger' trigger receives the event from the Broker TrChannel.",
+				Kind: model.Server,
+				Tags: map[string]string{
+					"http.method":      "POST",
+					"http.status_code": "202",
+					"http.host":        filterHost,
+					"http.path":        loggerTriggerPath,
+				},
+				Children: []tracinghelper.TestSpanTree{
+					{
+						Note: "21. Broker Filter for the 'logger' trigger sends the event to the logger pod.",
+						Kind: model.Client,
+						Tags: map[string]string{
+							"http.method":      "POST",
+							"http.status_code": "202",
+							"http.url":         fmt.Sprintf("http://%s/", loggerSVCHost),
```

Comment:
Can we remove the trailing `/`?

Reply:
No. This is doing an exact match against the values recorded in the trace. The trace records the trailing `/`, so the test needs it too.